core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -80,7 +80,18 @@
     protected final ReleasableLock readLock = new ReleasableLock(rwl.readLock());
     protected final ReleasableLock writeLock = new ReleasableLock(rwl.writeLock());
     protected volatile Throwable failedEngine = null;
-    protected volatile long lastWriteNanos = Long.MAX_VALUE; // no write yet!
+    /*
+     * on <tt>lastWriteNanos</tt> we use System.nanoTime() to initialize this since:
+     *  - we use the value for figuring out if the shard / engine is active so if we startup and no write has happened yet we still consider it active
```

Comment:
minor note: we consider shards inactive until the first indexing operation has happen, so I think this part is OK regardless of the change.